worker-api/src/main/java/com/hpe/caf/api/worker/NotIndendedTaskMessageForwardingEvaluator.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2015-2021 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hpe.caf.api.worker;
+
+import com.hpe.caf.api.Codec;
+import java.util.Map;
+
+/**
+ * A worker should implement this interface if it is capable of evaluating task messages that are not intended for it (because the
+ * worker's input queue is different from the target/to queue in the task message), and deciding whether they are to be discarded,
+ * executed or forwarded.
+ */
+public interface NotIndendedTaskMessageForwardingEvaluator
```

Comment:
I know we talked about having the new worker implement the existing `TaskMessageForwardingEvaluator` interface, but I couldn't see how to make that work for both the new worker, which does this:

https://github.com/JobService/worker-taskstowing/blob/SCMOD-12730/worker-taskstowing/src/main/java/com/microfocus/caf/worker/taskstowing/factory/TaskStowingWorkerFactory.java#L98

```
@Override
public TaskForwardingAction determineForwardingAction(
    final TaskMessage tm,
    final TaskInformation taskInformation,
    final boolean poison,
    final Map<String, Object> headers,
    final Codec codec,
    final JobStatus jobStatus,
    final WorkerCallback callback)
{
    return jobStatus == JobStatus.Paused
        ? TaskForwardingAction.Execute
        : TaskForwardingAction.Forward;
}
```
and the existing job tracking worker, which does this:

https://github.com/JobService/job-service/blob/develop/worker-jobtracking/src/main/java/com/hpe/caf/worker/jobtracking/JobTrackingWorkerFactory.java

```
@Override
public void determineForwardingAction(TaskMessage proxiedTaskMessage, TaskInformation taskInformation,
                                      Map<String, Object> headers, WorkerCallback callback) {

    ...
    LOG.debug("Forwarding task {}", proxiedTaskMessage.getTaskId());
    callback.forward(taskInformation, proxiedTaskMessage.getTo(), proxiedTaskMessage, headers);
}
```
    
One possible way would be to update the existing interface to return `TaskForwardingAction` instead of `void`, but in order to maintain the current behaviour of the job tracking worker, we would need to add a `NO_OP` value to `TaskForwardingAction`, that the job tracking worker could return instead of `void`:

```
@Override
public TaskForwardingAction determineForwardingAction(TaskMessage proxiedTaskMessage, TaskInformation taskInformation, Map<String, Object> headers, WorkerCallback callback) {
    ...
    callback.forward(taskInformation, proxiedTaskMessage.getTo(), proxiedTaskMessage, headers);
    return TaskForwardingAction.NO_OP;
}
```

Reply:
What you did in supporting a new interface and deprecating the old one is fine.